agent_api_http/src/v1/identity/keys/mod.rs

```diff
@@ -0,0 +1,165 @@
+use crate::IdentityContext;
+use agent_identity::managed_key::aggregate::{ManagedKey, SigningAlgorithm};
+use agent_identity::managed_key::command::ManagedKeyCommand;
+
+use agent_shared::handlers::{command_handler, query_handler};
+use axum::extract::{Json, State};
+use http_api_problem::ApiError;
+use hyper::StatusCode;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ManagedKeyDto {
+    #[serde(rename = "id")]
+    pub managed_key_id: String,
+    pub key_id: String,
+    pub alias: String,
+    pub signing_algorithm: Option<SigningAlgorithm>,
+}
+
+impl From<ManagedKey> for ManagedKeyDto {
+    fn from(value: ManagedKey) -> Self {
+        Self {
+            managed_key_id: value.managed_key_id,
+            key_id: value.key_id,
+            alias: value.alias,
+            signing_algorithm: value.signing_algorithm,
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Default)]
+#[serde(default, rename_all = "camelCase")]
+pub struct PostGenerateKey {
+    pub alias: String,
+    pub signature_algorithm: Option<SigningAlgorithm>,
+}
+
+#[axum_macros::debug_handler]
+pub(crate) async fn generate_key(
+    State(context): State<IdentityContext>,
+    Json(payload): Json<PostGenerateKey>,
+) -> Result<(StatusCode, String), ApiError> {
+    let signing_algorithm = payload.signature_algorithm.unwrap_or(SigningAlgorithm::ES256);
+
+    let managed_key_id = context
+        .key_generation_saga
+        .generate_key(payload.alias, signing_algorithm)
+        .await
+        .map_err(|e| {
+            tracing::error!("Saga failed: {:?}", e);
+            ApiError::new(StatusCode::INTERNAL_SERVER_ERROR)
+        })?;
+
+    Ok((StatusCode::CREATED, managed_key_id))
+}
+#[derive(Deserialize)]
+pub struct PostRemoveKey {
+    pub key_id: String,
+}
+
+pub(crate) async fn remove_key(
+    State(context): State<IdentityContext>,
+    Json(payload): Json<PostRemoveKey>,
+) -> Result<StatusCode, ApiError> {
+    let managed_key_id = query_handler(&payload.key_id, &context.state.query.managed_key)
```

Comment:
We cannot use `payload.key_id` for the `view_id` unfortunately because `key_id` is not the Aggregate identifier (`managed_key_id` is). 

So what we need is a method to map this the `key_id` to the corresponding `managed_key_id` and the only way of doing that currently is by calling `query_handler("all_managed_keys", &context.state.query.all_managed_keys)` and then `find` the correct key in the `AllManagedKeysView`. It would be nice if we had a generic method for this that would also work elsewhere but for now I would suggest you create a function that you can also reuse in the other endpoints that have the same problem (`rename_key_alias`, `set_signing_key`).